[TECH] Uniformisation des `scripts` des `package.json`

yannbertrand opened the pull request:
## :unicorn: Problème
Le champ `scripts` du `package.json` permet de partager des commandes shell qu'on exécute régulièrement en local ou dans la CI via `npm run` : 

```shell
$ npm run lint # execute toutes les tâches de lint d'un projet
```

Avec le temps, nos configurations se sont complexifiées et des choix différents ont été pris selon les projets.

Selon les projets sur lesquels on travaille, on peut être emmener à télécharger `npm-run-all` et/ou `concurrently` pour paralléliser l’exécution de commandes, résultant en plus de complexité lors de la maintenance (montée de version multiple).

## :robot: Propositions
- Suppression de scripts inutilisés
- Uniformisation de l'usage de `npm-run-all` via [`run-p`](https://github.com/mysticatea/npm-run-all/blob/HEAD/docs/run-p.md)
- Utilisation d'une seule même version d'`npm-run-all` (à la manière de certaines configs eslint). Nécessite `npm ci` à la racine
- Utilisation d'une seule même version de `eslint-plugin-i18n-json`
- Uniformisation des noms de script de lint sur tous les projets du monorepo
  * `lint` exécute tous les lint possibles en parallèle
  * `lint:fix` lance tous les autofix possibles en parallèle
  * `lint:js` et `lint:js:fix` pour le JavaScript
  * `lint:hbs` et `lint:hbs:fix` pour les templates front
  * `lint:scss` et `lint:scss:fix` pour les styles front
  * `lint:translations` et `lint:translations:fix` pour les traductions
- **Pix 1D** Lancement d'ESlint avec l'option de caching des autres projets
- **Mon Pix** Remise à jour du README.md et de la commande de lancement en se basant sur https://github.com/ember-cli/ember-new-output/blob/v4.12.1

## :rainbow: Remarques
Un `npm ci` à la racine du monorepo est nécessaire.

Sur plusieurs fronts, les scripts d'autofix apparaissent avec `lint:fix`, `lint:js:fix`, `lint:hbs:fix`, `lint:scss:fix` et `lint:translations:fix`.

Sur l'API, `lint:code` devient `lint:js` pour simplifier les choses (comme sur les fronts), c'est ce qui est concrètement fait.

Les `manual-lint:json` deviennent des `lint:translations`. On utilise partout le formateur eslint fourni par `eslint-plugin-i18n-json`.

Les `lint-fix` de l'API et de certif deviennent des `lint:fix`.

## :100: Pour tester
Vérifier que la CI passe.

Vérifier que les commandes de `lint` et de `lint:fix` passent.

Vérifier que les hooks de commit fonctionnent toujours.

yannbertrand committed:
sr: remove unused scripts

yannbertrand committed:
sr(1d): use npm-run-all rather than concurrently

yannbertrand committed:
sr(orga): reorder script alphabetically

yannbertrand committed:
sr: use shorter command to do parallel execution

yannbertrand committed:
sr: use root npm-run-all dependency

yannbertrand committed:
feat(1d): run eslint with content cache

yannbertrand committed:
feat: rename lint scripts to get more uniformity

API:
> lint-fix > lint:fix
> lint:code > lint:js
> lint:code:uncached > lint:js:uncached
> lint-fix:code > lint:js:fix
> lint-fix:translations > lint:translations:fix

Certif:
> lint-fix:translations > lint:translations:fix

yannbertrand committed:
feat: add lint fix scripts

pix-bot-github commented:
Une fois les applications déployées, elles seront accessibles via les liens suivants :
  * [App (.fr)](https://app-pr6433.review.pix.fr)
  * [App (.org)](https://app-pr6433.review.pix.org)
  * [Orga (.fr)](https://orga-pr6433.review.pix.fr)
  * [Orga (.org)](https://orga-pr6433.review.pix.org)
  * [Certif (.fr)](https://certif-pr6433.review.pix.fr)
  * [Certif (.org)](https://certif-pr6433.review.pix.org)
  * [Admin](https://admin-pr6433.review.pix.fr)
  * [API](https://api-pr6433.review.pix.fr/api/)
  * [Pix1D](https://1d-pr6433.review.pix.fr)

Les variables d'environnement seront accessibles via les liens suivants :
  * [scalingo front](https://dashboard.scalingo.com/apps/osc-fr1/pix-front-review-pr6433/environment)
  * [scalingo api](https://dashboard.scalingo.com/apps/osc-fr1/pix-api-review-pr6433/environment)

yannbertrand committed:
sr(mon-pix): prefer ember serve rather than its alias server

laura-bergoens commented:
pourquoi y'a t'il des changements dans les package-lock ?

yannbertrand commented:
> pourquoi y'a t'il des changements dans les package-lock ?

J'ai supprimé `npm-run-all` et `concurrently` des dépendances pour n'utiliser que le `npm-run-all` à la racine du monorepo.

laura-bergoens commented:
Alors je sais pas si c'est simple pour toi, mais penses-tu pouvoir faire la même petite passe sur pix-editor ? je pense qu'il y a des petits bidules qui peuvent être corrigés dans le sens de ta PR

laura-bergoens reviewed (approved):
Pour moi c'est bon, j'ai comparé le dernier run CI sur dev avec celui de ta PR, tout pareil.
![pr_yann_ok_01](https://github.com/1024pix/pix/assets/48727874/1f343b51-e2a3-48bf-8d02-6ec43a2dce2c)
![pr_yann_ok_02](https://github.com/1024pix/pix/assets/48727874/c2624372-b2e7-41c4-bedb-da6cb2403b8d)

aceol reviewed (approved):
Ca semble ok, merci pour le taff!

lego-technix commented:
Quand j'exécute `npm run lint` à la racine du monorepo j'obtiens la sortie ci-dessous, est-ce normal ?

```
ERROR: "lint:pix1d" exited with 127.
```

lego-technix commented:
Dans ce *super* travail d'uniformisation, ne pourrait-on pas aussi uniformiser les `manual-lint:json` et les transformer en `lint:translation`, ce qui serait aussi plus correct et plus métier ?

Et aussi pouvoir avoir un `lint:translations:fix` dans les apps Ember, comme dans l'api, serait idéal !

yannbertrand commented:
> Quand j'exécute `npm run lint` à la racine du monorepo j'obtiens la sortie ci-dessous, est-ce normal ?
> 
> ```
> ERROR: "lint:pix1d" exited with 127.
> ```

Merci pour le check, effectivement les `npm ci` sont nécessaires dans tous les sous projets pour pouvoir lancer le `lint` à la racine (c'est également le cas sur `dev`).

Personnellement je ne lance jamais ce script et la CI lance `lint:scripts`, `lint:yaml` et `test:root` uniquement. Je ne sais pas si d'autres lancent cette commande à la main mais au moins c'est uniforme ^^

lego-technix commented:
> > Quand j'exécute `npm run lint` à la racine du monorepo j'obtiens la sortie ci-dessous, est-ce normal ?
> > ```
> > ERROR: "lint:pix1d" exited with 127.
> > ```
> 
> Merci pour le check, effectivement les `npm ci` sont nécessaires dans tous les sous projets pour pouvoir lancer le `lint` à la racine (c'est également le cas sur `dev`).
> 
> Personnellement je ne lance jamais ce script et la CI lance `lint:scripts`, `lint:yaml` et `test:root` uniquement. Je ne sais pas si d'autres lancent cette commande à la main mais au moins c'est uniforme ^^

Je ne connaissais pas `lint:scripts`, alors que par contre je m'attendais à trouver à la racine du projet un npm script `lint`.

yannbertrand commented:
> > > Quand j'exécute `npm run lint` à la racine du monorepo j'obtiens la sortie ci-dessous, est-ce normal ?
> > > ```
> > > ERROR: "lint:pix1d" exited with 127.
> > > ```
> > 
> > 
> > Merci pour le check, effectivement les `npm ci` sont nécessaires dans tous les sous projets pour pouvoir lancer le `lint` à la racine (c'est également le cas sur `dev`).
> > Personnellement je ne lance jamais ce script et la CI lance `lint:scripts`, `lint:yaml` et `test:root` uniquement. Je ne sais pas si d'autres lancent cette commande à la main mais au moins c'est uniforme ^^
> 
> Je ne connaissais pas `lint:scripts`, alors que par contre je m'attendais à trouver à la racine du projet un npm script `lint`.

Oui je pense aussi que ça ne coûte pas grand chose de l'avoir. Par habitude je `lint` plutôt directement dans le projet sur lequel je bosse (je suis rarement multi projets).

Ça pose la question d'exécuter ces autres tâches de lint dans le `lint` global non ?

yannbertrand committed:
feat: run all lint and test tasks in root

yannbertrand commented:
> > > > Quand j'exécute `npm run lint` à la racine du monorepo j'obtiens la sortie ci-dessous, est-ce normal ?
> > > > ```
> > > > ERROR: "lint:pix1d" exited with 127.
> > > > ```
> > > 
> > > 
> > > Merci pour le check, effectivement les `npm ci` sont nécessaires dans tous les sous projets pour pouvoir lancer le `lint` à la racine (c'est également le cas sur `dev`).
> > > Personnellement je ne lance jamais ce script et la CI lance `lint:scripts`, `lint:yaml` et `test:root` uniquement. Je ne sais pas si d'autres lancent cette commande à la main mais au moins c'est uniforme ^^
> > 
> > 
> > Je ne connaissais pas `lint:scripts`, alors que par contre je m'attendais à trouver à la racine du projet un npm script `lint`.
> 
> Oui je pense aussi que ça ne coûte pas grand chose de l'avoir. Par habitude je `lint` plutôt directement dans le projet sur lequel je bosse (je suis rarement multi projets).
> 
> Ça pose la question d'exécuter ces autres tâches de lint dans le `lint` global non ?

J'ai mis à jour la PR pour ajouter l'exécution de ces tâches de lint et test supplémentaires. Pour info le `lint:docs` n'avait pas été lancé depuis un moment et retourne pas mal d'erreurs à cause de liens morts dans nos fichiers markdown.

yannbertrand committed:
sr: rename translations lint scripts to get more uniformity

J'ai exclu leur exécution de certains projets où ce lint ne passait pas, il faudra les corriger ailleurs, plus tard.

yannbertrand commented:
> Dans ce _super_ travail d'uniformisation, ne pourrait-on pas aussi uniformiser les `manual-lint:json` et les transformer en `lint:translation`, ce qui serait aussi plus correct et plus métier ?
> 
> Et aussi pouvoir avoir un `lint:translations:fix` dans les apps Ember, comme dans l'api, serait idéal !

Done. De la même manière que `npm-run-all`, j'ai rapatrié la dépendance `eslint-plugin-i18n-json` à la racine.

Il faudrait par la suite :
- [x] ajouter ce lint sur admin https://github.com/1024pix/pix/pull/6441
- [ ] corriger les lints de mon-pix, orga et admin
- [ ] ne pas tricher en ignorant les règles de lint :p
- [ ] revoir les configs de lint pour ne pas exécuter prettier sur les fichiers JSON
